rulesets/src/naming.ruleset.yml

```diff
@@ -52,19 +94,162 @@ rules:
         functionOptions:
           match: "^string$"
 
-  # fingerprint property usage and naming
-  sps-fingerprint-naming:
-    description: Rather than property names refering to the implementation for 'hash' or 'hashkey', you MUST use the property name 'fingerprint'.
-    message: "{{property}} is not using property name fingerprint."
+  ##### Standard Properties #####
+  sps-invalid-id-type:
+    description: id SHOULD use a data type of 'string'.
+    severity: warn
+    formats: [oas3]
+    given: '$..[?(@property === "id")].type'
+    then:
+      function: pattern
+      functionOptions:
+        match: "^string$"
+  
+  sps-mandate-abbreviations-identifier:
+    description: Use abbreviations instead of long form names, i.e. identifier SHOULD BE id.
+    severity: warn
+    formats: [oas3]
+    given: '$..properties.*~'
+    then:
+      function: pattern
+      functionOptions:
+        notMatch: "(^identifier([A-Z]|$)|.*Identifier([A-Z]|$))"
+
+  sps-mandate-abbreviations-reference:
+    description: Use abbreviations instead of long form names, i.e. reference SHOULD BE ref.
+    severity: warn
+    formats: [oas3]
+    given: '$..properties.*~'
+    then:
+      function: pattern
+      functionOptions:
+        notMatch: "(^reference([A-Z]|$)|Reference([A-Z]|$))"
```

Comment:
I think we should drop this as I can see a legit usage of "reference"... (different than id vs identifier)